packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -27,6 +27,12 @@ export interface VariableResolveOptions {
      * Used for resolving inputs, see https://code.visualstudio.com/docs/editor/variables-reference#_input-variables
      */
     configurationSection?: string;
+    // Holds variable-names to command id mappings (e.g. Provided by specific plugins / extensions)
+    // see "variables": https://code.visualstudio.com/api/references/contribution-points#contributes.debuggers
+    commandIdVariables?: Record<string, string>;
```

Comment:
Suggestion: Since the `Record<string, string>` type is passed around quite a lot in this PR, maybe give it an explicit interface to better convey its meaning? Something like:

```ts
/**
 * Holds variable-names to command id mappings (e.g. Provided by specific plugins / extensions)
 */
export interface CommandIdVariables {
    [id: string]: string
}
```